src/sentry_database.c

```diff
@@ -49,13 +50,20 @@ sentry__run_new(const sentry_path_t *database_path)
     run->run_path = run_path;
     run->session_path = session_path;
     run->lock = sentry__filelock_new(lock_path);
-    if (!run->lock || !sentry__filelock_try_lock(run->lock)) {
-        sentry__run_free(run);
-        return NULL;
+    if (!run->lock) {
+        goto error;
+    }
+    if (!sentry__filelock_try_lock(run->lock)) {
+        SENTRY_WARNF("failed to lock file \"%s\" (%s)", lock_path->path,
+            strerror(errno));
```

Comment:
Should we go the extra mile and use the thread-safe alternatives to `strerror()`?

Reply:
I don’t think it hurts, though we have a lock around `sentry_init`, and it is supposed to be called from the main thread during initialization, so maybe not worth it if its a large effort?

Reply:
It is not a massive effort, but `strerror_r()` behaves differently in different environments, so I would probably add a wrapper with the right `#ifdefs` to `sentry_util`. 

Interestingly while we do access `errno` in a couple of places, I haven't seen any conversions to readable strings for logging (though we also log `write_buffer_with_flags()`). What I am trying to get across here: while it might not be strictly necessary to have a thread-safe `errno` printer, I can imagine this to help with logging in other UNIXy code.